test_cli/test/initial_params.py

```diff
@@ -21,7 +21,10 @@ def main(argv=sys.argv):
     rclpy.init(args=argv)
 
     node = rclpy.create_node(
-        'initial_params_node', namespace='/', allow_undeclared_parameters=True)
+        'initial_params_node',
+        namespace='/',
+        allow_undeclared_parameters=True,
+        automatically_declare_initial_parameters=True)
```

Comment:
Doesn't seem like you need `allow_undeclared_parameters`, rclcpp only uses this:

https://github.com/ros2/system_tests/blob/9cb12dc22c853776c7a55a24e921bf1b96dec6b0/test_cli/test/initial_params.cpp#L26